pkg/capabilities/consensus/ocr3/median/median_aggregator.go

```diff
@@ -0,0 +1,256 @@
+package median
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"math"
+	"math/big"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/shopspring/decimal"
+	"google.golang.org/protobuf/proto"
+
+	ocrcommon "github.com/smartcontractkit/libocr/commontypes"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/datastreams"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+const (
+	// Aggregator outputs reports in the following format:
+	//   []Reports{FeedID []byte, RawReport []byte, Price *big.Int, Timestamp int64}
+	// Example of a compatible EVM encoder ABI config:
+	//   (bytes32 FeedID, bytes RawReport, uint256 Price, uint64 Timestamp)[] Reports
+	TopLevelListOutputFieldName = "Reports"
+	FeedIDOutputFieldName       = "FeedID"
+	RawReportOutputFieldName    = "RawReport"
+	PriceOutputFieldName        = "Price"
+	TimestampOutputFieldName    = "Timestamp"
+	RemappedIDOutputFieldName   = "RemappedID"
```

Comment:
@justinkaseman The actual implementation of this shouldn't have any data feeds specific concepts in it, and shouldn't make any assumptions about the shape of the observation.